### PR TITLE
feat(linter/promise): add `spec-only` rule

### PR DESCRIPTION
**Pull request by jelly**
Rule Detail:

[link](https://github.com/eslint-community/eslint-plugin-promise/blob/main/docs/rules/spec-only.md)

### CODE REVIEW

**Comment by jelly** [comment]
Not sure what:
```
---- size_asserts stdout ----
thread 'size_asserts' panicked at crates/oxc_linter/src/lib.rs:50:5:
assertion failed: std::mem::size_of::<RuleEnum>() == 16
```

Is about?

File: crates/oxc_linter/src/rules/promise/spec_only.rs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+use oxc_macros::declare_oxc_lint;
+use oxc_span::GetSpan;
+use rustc_hash::FxHashSet;
+
+use crate::{context::LintContext, rule::Rule, utils::PROMISE_STATIC_METHODS, AstNode};
+
+#[derive(Debug, Default, Clone)]
+pub struct SpecOnly {
+    allowed_methods: Option<FxHashSet<String>>,
+}
```
Comment (camc314) [inline crates/oxc_linter/src/rules/promise/spec_only.rs:12]: this should be boxed - see this for an example

https://github.com/oxc-project/oxc/blob/7a75e0f8a77006ef32fa0db45a329d961e70bdf3/crates/oxc_linter/src/rules/eslint/default_case.rs#L15-L29

Comment (camc314) [reply crates/oxc_linter/src/rules/promise/spec_only.rs:12]: this is also why you're seeing the test failures (
```
thread 'size_asserts' panicked at crates/oxc_linter/src/lib.rs:50:5:
assertion failed: std::mem::size_of::<RuleEnum>() == 16
```

Comment (jelly) [reply crates/oxc_linter/src/rules/promise/spec_only.rs]: Thanks, resolved.

**Review by DonIsaac — APPROVED** [review]

**Pull request merged**